pkg/medusa/reconcile.go

```diff
@@ -518,6 +524,52 @@ func StandaloneMedusaService(dcConfig *cassandra.DatacenterConfig, medusaSpec *a
 	return medusaService
 }
 
+func PurgeCronJob(dcConfig *cassandra.DatacenterConfig, medusaSpec *api.MedusaClusterTemplate, clusterName, namespace string, logger logr.Logger) *batchv1.CronJob {
+	purgeCronJob := &batchv1.CronJob{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      MedusaPurgeCronJobName(cassdcapi.CleanupForKubernetes(clusterName), dcConfig.SanitizedName()),
+			Namespace: namespace,
+		},
+		Spec: batchv1.CronJobSpec{
+			Schedule:                   "0 0 * * *",
+			Suspend:                    pointer.Bool(false),
+			SuccessfulJobsHistoryLimit: pointer.Int32(3),
+			FailedJobsHistoryLimit:     pointer.Int32(1),
+			JobTemplate: batchv1.JobTemplateSpec{
+				Spec: batchv1.JobSpec{
+					Template: corev1.PodTemplateSpec{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "k8ssandra-purge-backups",
+						},
+						Spec: corev1.PodSpec{
+							RestartPolicy:                 corev1.RestartPolicyOnFailure,
+							TerminationGracePeriodSeconds: pointer.Int64(30),
+							DNSPolicy:                     corev1.DNSClusterFirst,
+							SchedulerName:                 "default-scheduler",
```

Comment:
[nitpick] These are the default values, do we need to provide them explicitly?

Besides, it's not clear to me if the default scheduler is named `default-scheduler` in every distribution, I've seen references to `kube-scheduler` in some docs, so it might be better to leave it out.

Reply:
done